### PR TITLE
Improve getting transfer errors files

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: artifactory/commands/transferfiles/utils.go [hunk 1]
```diff
@@ -264,13 +267,22 @@ func getErrorsFiles(repoKey string, isRetry bool) (filesPaths []string, err erro
 		return []string{}, err
 	}
 
-	filesNames, err := biUtils.ListFiles(dirPath, false)
+	errorsFilesRegex := fmt.Sprintf(errorsFilesRegexFormat, repoKey)
+
+	regExp, err := regexp.Compile(errorsFilesRegex)
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:272]: ```regexp.Compile``` is an expensive operation. It is therefore better to it only once per CLI execution.

File: artifactory/commands/transferfiles/utils.go [hunk 2]
```diff
@@ -264,13 +267,22 @@ func getErrorsFiles(repoKey string, isRetry bool) (filesPaths []string, err erro
...
 	if err != nil {
 		return nil, err
 	}
 
-	for _, file := range filesNames {
-		if strings.HasPrefix(filepath.Base(file), repoKey) {
+	for _, file := range files {
+		matches := regExp.FindStringSubmatch(filepath.Base(file))
+		// Expecting a match and 4 groups.
+		if len(matches) == 5 {
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:285]: We're checking that the len is 5, but the above comment says `4 groups`. Is this a mistake?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/commands/transferfiles/utils.go (hunk 2 above)
Comment (RobiNino) [reply artifactory/commands/transferfiles/utils.go:285]: No, this function returns the whole match and the value of each group. I'll try rephrasing the comment above.

File: artifactory/commands/transferfiles/utils.go (hunk 1 above)
Comment (RobiNino) [reply artifactory/commands/transferfiles/utils.go:272]: This regexp contains the repo key, so it cannot be compiled once. As discussed, leaving as it is

**Pull request merged**